core-services/src/main/java/org/zalando/nakadi/partitioning/MetadataRandomPartitioner.java

```diff
@@ -0,0 +1,45 @@
+package org.zalando.nakadi.partitioning;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.zalando.nakadi.cache.EventTypeCache;
+import org.zalando.nakadi.service.timeline.TimelineService;
+
+import java.util.List;
+import java.util.Random;
+
+@Component
+public class MetadataRandomPartitioner {
```

Comment:
there is the class https://github.com/zalando/nakadi/blob/master/core-services/src/main/java/org/zalando/nakadi/partitioning/RandomPartitionStrategy.java, can we reuse it ?

Reply:
I have postponed this decision until the Publishing part for Avro is in place. Suppose we want to reuse those Partition Strategy implementations. In that case, we need to access Event Metadata or Event payload. Having the publishing part in place would make the decision process more straightforward and safer so that we can use all the partition strategies in JSON, Avro, and Metadata event types.

Reply:
👍 then we will have to remove it once publisher is ready to work with partitioning

Reply:
Since we extended the scope, I think the comment is resolved with the current implementation